### DISCUSSION TITLE
Git Repository No Longer Works

### DISCUSSION THREAD

**Discussion by andrew-dixon** [phabricator-docker/phabricator#44] [Q&A] [answered]
Since I added the daemon container, as per issue #41, the git repos on Phabricator no longer work correctly. They still appear to be updating, as commits appear on tickets, however, if you click on the commit it takes you to the commit with a warning message at the top that reads:

```
This commit no longer exists in the repository. It may have been part of a branch which was deleted.
```

And there is not code shown.

Also, if you try to browse the repos via the Diffusion section you get the error message:

```
Unhandled Exception ("CommandException")	
Command failed with error #128!
COMMAND
git log --skip 0 --max-count 15 '--format=%H:%P' e960201b68da2a886473759154119743052a4dbd --

STDOUT
(empty)

STDERR
fatal: bad object e960201b68da2a886473759154119743052a4dbd
```

If I start the daemon on the Phabricator container again and wait a couple of minutes it all starts working again. So, what have I got mis-configured here? Is there something extra I need in the configuration on the daemon container for the repos?

Thanks.

Andrew.

**Comment by davidbarratt** [comment, marked as answer]
hmm, I don't use the repos (I only use the task management) so you're in uncharted waters. :) Appreciate you testing it out. 

Maybe both the Phabricator and the Daemon need the same volume mounts in the same location? I imagine maybe the deamon is might be caching the repos in the database and it would need access to them.

In other words, both the Phabricator and the Daemon should have basically the exact same config, if you aren't mounting the repos somewhere you should start doing that so they can be persisted and shared between containers. 

Does that help?

**Reply by andrew-dixon** [reply]
@davidbarratt That makes sense, I'll give a try and report back.

**Reply by andrew-dixon** [reply]
I've tried this now and it works. So, for anyone else wanting to do it, this is what I did:

1) Created a volume in docker.
2) Connected that volume to both the docker phabricator and daemon container linked to `/var/repo` on the container.
3) Once linked, opened the console on he Phabricator container and ran `./bin/repository update R1` where R1 is the repository number in Phabricator. I have two repos so had to run `./bin/repository update R2` as well.
